crates/misc/dotnet/docs/articles/intro.md

```diff
@@ -0,0 +1,214 @@
+# Introduction to Wasmtime for .NET
+
+[Wasmtime](https://github.com/bytecodealliance/wasmtime) is a standalone runtime capable of executing [WebAssembly](https://webassembly.org/) outside of a web browser.
+
+Wasmtime for .NET is a .NET API for Wasmtime.  It enables .NET developers to easily instantiate and execute WebAssembly modules.
+
+For this tutorial, we will create a WebAssembly module from a program written in Rust and use that WebAssembly module from a .NET Core 3.0 application.
+
+# Creating a simple WebAssembly module
+
+One of the reasons why WebAssembly is so exciting is that [many languages are able to target WebAssembly](https://github.com/appcypher/awesome-wasm-langs).  This means, for example, a plugin model based on WebAssembly could enable developers to write sandboxed, cross-platform plugins in any number of languages.
+
+Here I've decided to use [Rust](https://www.rust-lang.org/) for the implementation of the WebAssembly module.  Rust is a modern systems programming language that can easily target WebAssembly.
+
+If you wish to skip creating the WebAssembly module, download the [prebuilt WebAssembly module](https://raw.githubusercontent.com/bytecodealliance/wasmtime/master/crates/misc/dotnet/docs/wasm/intro/hello.wasm) from this tutorial, copy it to your .NET project directory, and continue from the _[Using the WebAssembly module from .NET](#using-the-webassembly-module-from-net)_ section.
+
+## Installing a Rust toolchain
+
+To get started with Rust, install [rustup](https://rustup.rs/), the manager for Rust toolchains.
+
+This will install both a `rustup` command and a `cargo` command (for the active Rust toolchain) to your PATH.
+
+## Installing the WebAssembly target
+
+To target WebAssembly with the active Rust toolchain, install the WebAssembly [target triple](https://forge.rust-lang.org/release/platform-support.html):
+
+```text
+rustup target add wasm32-unknown-unknown
+```
+
+## Creating the Rust project
+
+Create a new Rust library project named `hello`:
+
+```text
+cargo new --lib hello
+cd hello
+```
+
+To target WebAssembly, the library needs to be built as a `cdylib` (dynamic library) rather than the default of a static Rust library.  Add the following to the `Cargo.toml` file in the project root:
+
+```toml
+[lib]
+crate-type = ["cdylib"]
+```
+
+## Implementing the WebAssembly code
+
+The WebAssembly implementation will import a `print` function from the host environment and pass it a string to print.  It will export a `run` function that will invoke the imported `print` function.
+
+Replace the code in `src/lib.rs` with the following Rust code:
+
+```rust
+extern "C" {
+    fn print(address: i32, length: i32);

```

Comment:
Does the C# extension support WASI right now? If so it may be best to give a slightly different example,  although if not we should open an issue to add support too :)

Reply:
It doesn't, as the C API doesn't support it and there's no way to enable it currently.

That is the motivation for me opening https://github.com/WebAssembly/WASI/issues/149.